C++/API/tests.cpp

```diff
@@ -1,74 +1,61 @@
-#include <iostream>                  // std::cout, std::cin
-#include <string>                    // std::string, std::to_string;
-#include "include/API.h"             // API class
-
-using std::cin;
-using std::cout;
-using std::endl;
-using std::string;
-using std::to_string;
+#include "include/API.h"
 
 // For picojson
 using namespace picojson;
 
-/*
- * This file requires Boost. Please make sure to run the following command before compiling:
+/* This file requires Boost. Please make sure to run the following command
+ * before compiling:
  * sudo apt-get install libboost-test-dev
- * The above command installs Boost into your development environment.
- *
+ * The above command will install Boost into your development environment.
  */
 #define BOOST_TEST_DYN_LINK
 #define BOOST_TEST_MODULE Main
 #include <boost/test/unit_test.hpp>
 
-
-/*
-  This file will run a series of tests on the C++ API
-  The following methods should be tested:
-
-  iSENSE()
-  iSENSE(string proj_ID, string proj_title, string label, string contr_key)
-  set_project_all(string proj_ID, string proj_title, string label, string contr_key)
-  push_back(string)
-  push_back(vector<string>)
-  clear_data()
-  format_data_upload()
-  get_check_user()
-  post_json_key()
-  post_json_email()
-
-*/
+/* This file will run a series of tests on the C++ API
+ * The following methods are tested:
+ *
+ * get_check_user()
+ * get_project_fields()
+ * get_datasets_and_mediaobjects()
+ * get_projects_search()
+ * get_dataset()
+ * get_dataset_ID()
+ * get_field_ID()
+ * post_json_email()
+ * post_json_key()
+ * append_email_byID()
+ * append_email_byName()
+ * append_key_byID()
+ * append_key_byName()
+ *
+ */
 
 
 /*
  * This is a derived class to quickly test the append_byID functions.
  * It is derived from iSENSE, and by doing this I can create a public function
- * which can access the two append_byID functions publicly. (for testing these functions)
- * Users would have to derive the class to do the same - simply using iSENSE, the
- * append_byID functions will be protected and unaccessable (since they should be using
- * the "append_byName, using dataset names rather than dataset IDs)
- * 
+ * which can access the two append_byID functions publicly.
+ * Users would have to derive the class to do the same - simply using iSENSE,
+ * the append_byID functions will be protected and unaccessable (since they
+ * should be using the "append_byName, using dataset names rather than IDs)
  */
-class Test: public iSENSE
-{
-public:
-  void check_set_dataset_ID(string proj_dataset_ID)
-  { 
+class Test: public iSENSE {

```

Comment:
I haven't added that yet, that will probably be the next thing I do, after I add a method for error checking HTTP codes and empty objects.